functions/Install-DbaFirstResponderKit.ps1

```diff
@@ -109,123 +108,167 @@ function Install-DbaFirstResponderKit {
         if (-not $DbatoolsData) {
             $DbatoolsData = [System.IO.Path]::GetTempPath()
         }
-
-        $url = "https://github.com/BrentOzarULTD/SQL-Server-First-Responder-Kit/archive/$Branch.zip"
-
         $temp = [System.IO.Path]::GetTempPath()
-        $zipfile = Join-Path -Path $temp -ChildPath "SQL-Server-First-Responder-Kit-$Branch.zip"
-        $zipfolder = Join-Path -Path $temp -ChildPath "SQL-Server-First-Responder-Kit-$Branch"
-        $FRKLocation = "FRK_$Branch"
-        $LocalCachedCopy = Join-Path -Path $DbatoolsData -ChildPath $FRKLocation
+        if ($Branch -eq 'master') {
+            $url = 'https://api.github.com/repos/BrentOzarULTD/SQL-Server-First-Responder-Kit/releases/latest'
+            $zipFile = Join-Path -Path $temp -ChildPath "SQL-Server-First-Responder-Kit-latest.zip"
+        } elseif ($Branch -eq 'dev') {
+            $url = "https://github.com/BrentOzarULTD/SQL-Server-First-Responder-Kit/archive/dev.zip"
+            $zipfile = Join-Path -Path $temp -ChildPath "SQL-Server-First-Responder-Kit-dev.zip"
+        } else {
+            Write-Message -Level Warning -Message "Unknown value provided for Branch parameter"
+        }
+
         if ($LocalFile) {
             if (-not(Test-Path $LocalFile)) {
-                Stop-Function -Message "$LocalFile doesn't exist"
-                return
+                if ($PSCmdlet.ShouldProcess($LocalFile, "File does not exists, returning to prompt")) {
+                    Stop-Function -Message "$LocalFile doesn't exist"
+                    return
+                }
             }
             if (-not($LocalFile.EndsWith('.zip'))) {
-                Stop-Function -Message "$LocalFile should be a zip file"
-                return
+                if ($PSCmdlet.ShouldProcess($LocalFile, "File is not a zip file, returning to prompt")) {
+                    Stop-Function -Message "$LocalFile should be a zip file"
+                    return
+                }
             }
         }
 
         if ($Force -or -not(Test-Path -Path $LocalCachedCopy -PathType Container) -or $LocalFile) {
             # Force was passed, or we don't have a local copy, or $LocalFile was passed
             if ($zipfile | Test-Path) {
-                Remove-Item -Path $zipfile -ErrorAction SilentlyContinue
-            }
-            if ($zipfolder | Test-Path) {
-                Remove-Item -Path $zipfolder -Recurse -ErrorAction SilentlyContinue
+                if ($PSCmdlet.ShouldProcess($zipfile, "File found, dropping $zipfile")) {
+                    Remove-Item -Path $zipfile -ErrorAction SilentlyContinue
+                }
             }
 
-            $null = New-Item -ItemType Directory -Path $zipfolder -ErrorAction SilentlyContinue
+            # $null = New-Item -ItemType Directory -Path $zipfolder -ErrorAction SilentlyContinue
```

Comment:
seems like this could be the issue? if it was creating the folder before, maybe that is some user or OS setting. Why did it pick the dbatools path tho, weird.

Reply:
It always used the dbatools path

Reply:
We bring the file down to the module path and then end up copying it over to the dbatools path within a user's profile as the "local cache"